src/hooks/useBreakpoint.tsx

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useState } from "react";
+import { BreakpointKey, Breakpoints } from "../Css";
+
+/**
+ * A React hook to return a record of responsive breakpoints that updates on resize.
+ *
+ * @example
+ * const { breakpoints } = useBreakpoint();
+ * if(breakpoints.mdAndDown) {...do something cool}
+ */
+
+type BreakpointsType = Record<BreakpointKey, boolean>;
+
+export function useBreakpoint(): { breakpoints: BreakpointsType } {
+  const [breakpoints, setBreakpoints] = useState({} as BreakpointsType);
+
+  useEffect(() => {
+    function handleResize() {
+      // Set breakpoints on resize
+      let bps = {} as BreakpointsType;
+      Object.entries(Breakpoints).forEach(([name, bp]) => {
+        bps[name as keyof BreakpointsType] = window.matchMedia(bp.replace("@media screen and ", "")).matches;
+      });
+      setBreakpoints(bps);
+    }
+    window.addEventListener("resize", handleResize);
```

Comment:
@bdow - checked our the  `useResizeObserver` but it needs a provided target which didnt feel as clean as just watching for a resize window event.

Reply:
I'm fine with that. Curious though, could the "window" or "body" be the target?